### PR TITLE
Fix text field generator with explicit value.

### PR DESCRIPTION
**Pull request by gbourass**
# Fix text field generator with explicit value.

## Description

When calling GenerateInput for a text field with `isExplicitValue` = true, if the root model has a property with the same name, the `value` parameter is ignored and the context is used regardless.

{Detail}

Fixes #{bug number} (in this specific format)

### CODE REVIEW

**Comment by adityamandaleeka** [comment]
@gbourass I don't believe this is a typo. The `attributeValue` is being initialized and used intentionally here in the case where `useViewData` isn't set. But it sounds like that behavior isn't what you expect?

**Comment by gbourass** [comment]
@adityamandaleeka Thanks for your reply. I believe you are right: the usage of attributeValue seems intentional now that I read it again.
However the code doesn't handle all the cases properly.
Please consider the following:
image
If useViewData is false and isExplicitValue is true.
At line 1357, the attributeValue is resolved with the partial fullName giving the wrong value.
Then at line 1358, since attributeValue is not null, the ??= operator is ignored so useViewData is not checked.
In other words, even if useViewData is false, it always resolve with the view data.
I would expect that if useViewData is false and isExplcitValue is true, then the method parameter value is what is set in the value attribute of the tagBuilder.
My current fix is wrong, but I believe I should make a new one that uses GetModelStateValue ONLY if useViewData is set.

**Comment by adityamandaleeka** [comment]
@mitchdenny PTAL when you have time.

**Comment by mkArtakMSFT** [comment]
@javiercn reassigning this to you to take a look.

**Comment by javiercn** [comment]
@gbourass It's a bit hard to determine what is going on here.

For us to consider a fix, we suggest you start with an issue explaining the scenario, the observed behavior and the expected behavior. 

For us to accept a PR we need to have a solid understanding of the issue, as otherwise we risk breaking other scenarios, and we need to see the PR include the relevant tests to provide confidence in the change.

**Comment by gbourass** [comment]
@javiercn I totally agree. I initially thought it was a quick fix (so I created a PR), but it is actually really complex with many possible side effects.
I will need to create bug and provide a full code example of how to reproduce the issue to make it simple to reproduce.

**Comment by gbourass** [comment]
Created repo to reproduct the issue: https://github.com/gbourass/dotnet-aspnetcore-bug-52150-text-field-rendering-post-conflict-demo
Someone else entered a similar issue in the meantime: https://github.com/dotnet/aspnetcore/issues/52150
I added the information in this issue.
